src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,534 @@
+            raise Exception("Failed to create \
+                  aggregate {}".format(aggregate_name))
+
+    def set_aggregate_metadata(self, aggregate_id, aggregate_metadata):
+        try:
+            logger.info("Setting aggregate metadata")
+            for metadata in aggregate_metadata:
+                self.aggregate_obj.set_metadata(aggregate_id, metadata)
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to set aggregate metadata \
+                  for aggregate with ID {}".format(aggregate_id))
+
+    def add_hosts_to_aggregate(self, aggregate_id, host_name_list):
+        try:
+            logger.info("Adding host to aggregate")
+            for host in host_name_list:
+                self.aggregate_obj.add_host(aggregate_id, host)
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to add hosts {} to aggregate \
+                  with ID {}".format(host_name_list, aggregate_id))
+
+    def get_dell_compute_nodes_hostnames(self):
+        try:
+            logger.info("Getting compute node hostnames")
+
+            # Create host object
+            host_obj = hosts.HostManager(self.nova)
+
+            # Get list of dell nfv nodes
+            dell_hosts = []
+            for host in host_obj.list():
+                if "dell-compute" in host.host_name:
+                    dell_hosts.append(host.host_name)
+            return dell_hosts
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to get the Dell Compute nodes.")
+
+    def configure_dell_aggregate(self,
+                                 aggregate_name,
+                                 aggregate_metadata,
+                                 host_name_list=None):
+        try:
+            logger.info("Configuring dell aggregate")
+
+            # Create aggregate
+            aggregate_id = self.create_aggregate(aggregate_name)
+
+            # Set aggregate metadata
+            self.set_aggregate_metadata(aggregate_id, aggregate_metadata)
+
+            if host_name_list:
+                # Add hosts to aggregate
+                self.add_hosts_to_aggregate(aggregate_id, host_name_list)
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to configure \
+                            dell aggregate {}.".format(aggregate_name))
+
+    def create_flavor(self, flavor_name):
```

Comment:
We already have a "create_flavors" method that does all of the work below.  We really shouldn't have 2 different pieces of code for doing the same thing.  I think i would modify this method so that it accepts memory, disk, cpus, and id, then modify the create_flavors method to call into this one.  Probably should rename create_flavors to create_default_flavors or something like that.

Reply:
It seems fine to me. Meanwhile, i would like to put this on hold, as there is a discussion on-going between you, gael and the rest of the team to decide the direction we'd like to take with respect to the flavors going forward. Besides, we need to set additional metadata to flavors used for CSP Profile.